mantidproject/mantid
# Fix segmentation fault when folding the pick tab in the instrument viewer

**Description of work.**
This is an attempt to solve the issues with folding the pick tab with the miniplot expanded.
It consists on checking if the tab (and thus the miniplot) is visible before any attempt to replot it, so the replot doesn't try to reach an object which doesn't exist anymore. 

**NB** : This PR solves half the issue - namely the segfault, but doesn't take care of the error thrown by numpy. My best guess is that this error is caused by matplotlib trying to resize the plot to a non-existent size, which leads to trouble inside of numpy.
But despite my efforts, I couldn't manage to catch it or even discover where this resizing takes place.



**To test:**
As usual in the instrument viewer, not much testing infrastructure is available so we rely on manual testing.

- Load any file and open the instrument viewer
- Go to the pick tab, make sure the miniplot is expanded - even if it shows nothing 
- Resize the tab so that it disappears, by dragging the border to the left, and hover the mouse over the instrument
- See that it only produces an error, but doesn't crash Mantid


Fixes #32432 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
